ts/features/fci/screens/valid/FciDocumentPreviewScreen.tsx

```diff
@@ -1,16 +1,152 @@
+  const dispatch = useIODispatch();
+
+  React.useEffect(() => {
+    const docUrl = pipe(
+      props.route.params,
+      O.fromNullable,
+      O.chainNullableK(p => p.documentUrl),
+      O.getOrElse(() => "")
+    );
+
+    if (docUrl.length > 0) {
+      setDocumentUrl(docUrl);
+    } else if (documents.length > 0) {
+      setShowDocNavBar(true);
+      setDocumentUrl(documents[currentDoc].url);
+    }
+  }, [currentDoc, documents, props.route.params]);
+
+  const customGoBack: React.ReactElement = (
+    <TouchableDefaultOpacity
+      onPress={() => {
+        if (showDocNavBar) {
+          dispatch(fciShowSignedDocumentsEndRequest());
+        } else {
+          dispatch(fciDownloadPreviewClear({ path: fciDownloadPath }));
+        }
+      }}
+      accessible={true}
+      accessibilityLabel={I18n.t("global.buttons.back")}
+      accessibilityRole={"button"}
+    >
+      <IconFont name={"io-back"} style={{ color: IOColors.bluegrey }} />
+    </TouchableDefaultOpacity>
+  );
+
+  const onPrevious = () => {
+    pipe(
+      currentDoc,
+      O.fromNullable,
+      O.chain(doc => (doc > 0 ? O.some(doc - 1) : O.none)),
+      O.map(setCurrentDoc)
+    );
+  };
+
+  const onNext = () => {
+    pipe(
+      currentDoc,
+      O.fromNullable,
+      O.chain(doc => (doc < documents.length - 1 ? O.some(doc + 1) : O.none)),
+      O.map(setCurrentDoc)
+    );
+  };
+
+  const renderDocumentsNavigationBar = () => (
+    <DocumentsNavigationBar
+      titleLeft={I18n.t("features.fci.documentsBar.titleLeft", {
+        currentDoc: currentDoc + 1,
+        totalDocs: documents.length
+      })}
+      titleRight={I18n.t("features.fci.documentsBar.titleRight", {
+        currentPage,
+        totalPages
+      })}
+      iconLeftColor={currentDoc === 0 ? IOColors.bluegreyLight : IOColors.blue}
+      iconRightColor={
+        currentDoc === documents.length - 1
+          ? IOColors.bluegreyLight
+          : IOColors.blue
+      }
+      onPrevious={onPrevious}
+      onNext={onNext}
+      disabled={false}
+      testID={"FciDocumentsNavBarTestID"}
+    />
+  );
 
-  return <DocumentViewer documentUrl={documentUrl} />;
+  return (
+    <BaseScreenComponent
+      goBack={true}
+      customGoBack={customGoBack}
+      contextualHelp={emptyContextualHelp}
+      headerTitle={I18n.t("features.mvl.details.attachments.pdfPreview.title")}
+    >
+      <SafeAreaView
+        style={styles.container}
+        testID={"FciDocumentPreviewScreenTestID"}
+      >
+        {showDocNavBar && renderDocumentsNavigationBar()}
+        {pipe(documentUrl, S.isEmpty) === false && (
+          <DocumentViewer
+            onLoadComplete={(pageCount: number) => {
+              setTotalPages(pageCount);
+            }}
+            onPageChanged={(page: number) => {
+              setCurrentPage(page);
+            }}
+            documentUrl={
+              pipe(documentUrl, S.isEmpty) === false
```

Comment:
Strange check because we already checked on line 132.

Reply:
Can you propose a solution?

Reply:
I have removed this check. This is outdated.